voluncheer/opportunityboard/views/postanopportunity.py

```diff
@@ -47,14 +51,18 @@ def update_an_opportunity(request, opportunity_id):
     if user.is_organization:
         organization_profile = Organization.objects.get(pk=user)
     if request.method == "POST":
-        form = PostAnOpportunityForm(request.POST, request.FILES)
+        form = PostAnOpportunityForm(request.POST, request.FILES, instance=opportunity_to_update)
         if form.is_valid():
             if "delete" in request.POST:
                 form.delete(opportunity_id)
             else:
-                form.update(opportunity_id)
+                form.save()
         else:
-            print(form.errors.as_data())
```

Comment:
We shouldn't render if the form is invalid. We should throw an error right?

Reply:
If we don't render, the user will not see the error message and will not know why the opportunity failed to be created.